### PR TITLE
Fix infinite range subscript examples

### PR DESCRIPTION
**Pull request by taboege**
cf. rakudo/rakudo#2872 and perl6/roast#536

### CODE REVIEW

File: doc/Language/list.pod6 [hunk 1]
```diff
@@ -480,14 +480,17 @@ which is selecting the 4 to 6th element from the three  first dimensions (C<^3>)
 A L<C<Range>|/type/Range> is a container for a lower and an upper boundary.
 Generating a slice with a C<Range> will include any index between those bounds,
 including the bounds. For infinite upper boundaries we agree with
-mathematicians that C<Inf> equals C<Inf-1>.
+mathematicians that C<Inf> equals C<Inf-1>. Slicing with an infinite Range
+will L<truncate|/language/subscripts#Truncating_slices>.
```
Comment (JJ) [inline doc/Language/list.pod6]: Not clear the relationship between the two sentences here. If it's "so,..." they the example contradicts them. it's only the combination of -1 and using the ^ which truncates the range. How does that follow from the two sentences?

File: doc/Language/list.pod6 [hunk 2]
```diff
@@ -480,14 +480,17 @@ which is selecting the 4 to 6th element from the three  first dimensions (C<^3>)
...
 
     my @a = 1..5;
     say @a[0..2];     # OUTPUT: «(1 2 3)␤»
     say @a[0..^2];    # OUTPUT: «(1 2)␤»
     say @a[0..*];     # OUTPUT: «(1 2 3 4 5)␤»
-    say @a[0..^*];    # OUTPUT: «(1 2 3 4)␤»
+    say @a[0..^*];    # OUTPUT: «(1 2 3 4 5)␤»
     say @a[0..Inf-1]; # OUTPUT: «(1 2 3 4 5)␤»
+    say @a[0..*-1];   # OUTPUT: «(1 2 3 4 5)␤»
+    say @a[0..^*-1];  # OUTPUT: «(1 2 3 4)␤»
```
Comment (JJ) [inline doc/Language/list.pod6]: Infinite minus one minus one is still infinite. How does this apply here?

File: doc/Language/list.pod6 (hunk 1 above)
Comment (taboege) [reply doc/Language/list.pod6]: There is no causal relationship between the two. I just found the information about truncation was missing from this discussion.

To fix this properly, without just removing it again, maybe this entire section about Range slicing should be taken off the List context section there and instead go into http://docs.perl6.org/language/subscripts#Slices. What do you think?

File: doc/Language/list.pod6 (hunk 2 above)
Comment (taboege) [reply doc/Language/list.pod6]: It doesn't apply. `1..*` is the same as `1..Inf` with or without the caret. But `1..*-1` suddenly is a WhateverCode that generates a Range whose upper boundary is filled in by the subscript operator with the list's .elems. So subtracting 1 twice indeed loses the last value. The interesting point is that the infinite ranges `1..*` and `1..^*` behave as if they were also WhateverCode, making the use of `*` strangely consistent. I take it that this (up to the strangely consistent part) should be explained on the page, too. Correct?

File: doc/Language/list.pod6 (hunk 1 above)
Comment (JJ) [reply doc/Language/list.pod6]: Maybe you could move it up in the paragraph, in the sentence starting with "Generating a slice"

File: doc/Language/list.pod6 (hunk 2 above)
Comment (JJ) [reply doc/Language/list.pod6]: Yep, probably. Clarifying that would really be helpful.

**Comment by taboege** [comment]
I pushed another commit in front, in addition to (hopefully) addressing your remarks. That additional commit adds to the Range documentation a paragraph about how WhateverCode is pulled through the .. operator, to which I refer in the list subscripting docs.

**Pull request merged**